### DISCUSSION TITLE
What is the nextJS paradigm for handling user authorization?

### DISCUSSION THREAD

**Discussion by patrick99e99** [vercel/next.js#70112] [Help]
### Summary

I am new to nextJS and have been tasked to integrate okta authentication to an existing app. When I am looking at this app, it is doing the authorization check inside the component, which does not seem like the right place to me. It is basically doing a useEffect, and within that, checking the store for a user object, if it's not there doing router.push() to go to the login page.

Is this the convention?

I was initially thinking middleware was the way to do this, but then I saw I could not access the store select/dispatch hooks inside middleware functions... So, then I am thinking maybe the _app component, which currently is doing:

      <Provider store={store}>
        <StyledEngineProvider injectFirst>
          <ThemeProvider theme={darkTheme}>
            <CssBaseline />
            <AppLoader />
            <Component {...pageProps} />
            <GlobalSnackbar />
            <ReduxRouter />
          </ThemeProvider>
        </StyledEngineProvider>
      </Provider>
Maybe I should add a <Auth /> after ` and have that be a component that does this behavior?

Or is there a better way to handle this?

All that really needs to happen is:

* does a user exist in the store, if so, let it proceed and api calls will 401 if their token is invalid
* if no user exists in the store, call the current user api endpoint (auth token in cookie), if it's successful put the user in the store and continue on
* if the current user api 401s, assume no auth token or it's expired, therefore redirect to external okta url

But my question ultimately is, where is the "correct" place in nextJS code land for this to happen?

**Comment by Juneezee** [comment]
I would recommend you to use Auth.js. You can use the out-of-the-box Okta Provider. See https://authjs.dev/getting-started/providers/okta
